### PR TITLE
Solution for PlaneBuilder

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        //Test Plane
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java:5]: please remove commented code before pushing

the purpose of your code should be clear without additional explanation

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public static void main(String[] args) {
+        //Test Plane
+        Plane.PlaneBuilder planeBuilder = new Plane.PlaneBuilder("Boing");
+        Plane airbus = planeBuilder.setEngines(3)
+                .setSeats(130)
+                .setWheels(14)
+                .setWings(2)
+                .setWindows(150)
+                .build();
+        System.out.println(airbus.toString());
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java:13]: toString() will be called automatically
```suggestion
        System.out.println(airbus);

```

File: src/main/java/core/basesyntax/Plane.java [hunk 3]
```diff
@@ -7,5 +7,102 @@
  */
 
 public class Plane {
+    private final String name;
+    private final int seats;
+    private final int wheels;
+    private final int wings;
+    private final int engines;
+    private final int windows;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Plane.java:15]: what is the purpose of making these fields final here?

File: src/main/java/core/basesyntax/Plane.java [hunk 4]
```diff
@@ -7,5 +7,102 @@
...
 
+    public Plane(PlaneBuilder planeBuilder) {
+        this.name = planeBuilder.name;
+        this.seats = planeBuilder.seats;
+        this.wheels = planeBuilder.wheels;
+        this.wings = planeBuilder.wings;
+        this.engines = planeBuilder.engines;
+        this.windows = planeBuilder.windows;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Plane.java:23]: there is no need to use `this` keyword here

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Main.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/Main.java:5]: Fixed.

File: src/main/java/core/basesyntax/Main.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/Main.java:13]: Fixed.

File: src/main/java/core/basesyntax/Plane.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/Plane.java:15]: I was thinking these fields are more protected if make them final, and little bit more difficult to change them so I tried to put final and to build Builder for this type of field. Anyway, I changed them to private.

File: src/main/java/core/basesyntax/Plane.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/core/basesyntax/Plane.java:23]: Removed.

**Pull request closed without merging**